fix: serde and packable on empty

benesjan opened the pull request:
Fixes #17586

Forgot to handle the empty struct case when implementing the derivation functionality. This PR fixes it.

benesjan commented:
* **#18584** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18584?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18584?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

mverzilli commented on noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr:
```diff
@@ -228,14 +228,20 @@ pub(crate) comptime fn derive_deserialize(s: TypeDefinition) -> Quoted {
+    // The following will give us:
+    // <type_of_struct_member_1 as Deserialize>::N + <type_of_struct_member_2 as Deserialize>::N + ...
+    // (or 0 if the struct has no members)
+    let right_hand_side_of_definition_of_n = if params.len() > 0 {
```
nitpick: maybe now that the `if` expression has three branches it is better suited for a `match`?

benesjan replied:
There is no `match` in Noir 😄

mverzilli replied:
Sads :cry:

mverzilli reviewed (approved):
Always a pleasure to review a one file PR :D

Only comment is we seem to be checking multiple times the same cases (params>1, params==1, params==0) so it would be nice to reduce that duplication, but I'm not sure whether that's possible in Noir and more specifically in Noir macros

benesjan commented:
> Only comment is we seem to be checking multiple times the same cases

Yes, there is a large overlap between serde and packable. I think we had a plan to eventually merge the Serialize and Deserialize traits into one trait and at that point they would become even more similar. So would wait until we do that to bother with this.

benesjan committed:
fix: serde and packable on empty

Fixes #17586

Forgot to handle the empty struct case when implementing the derivation functionality. This PR fixes it.
